rust/kernel/module_param.rs

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Types for module parameters.
+//!
+//! C header: [`include/linux/moduleparam.h`](../../../include/linux/moduleparam.h)
+
+use core::fmt::Write;
+
+/// Types that can be used for module parameters.
+///
+/// Note that displaying the type in `sysfs` will fail if
+/// [`alloc::string::ToString::to_string`] (as implemented through the
+/// [`core::fmt::Display`] trait) writes more than `kernel::PAGE_SIZE`
+/// bytes (including an additional null terminator).
+///
+/// [`alloc::string::ToString::to_string`]: https://doc.rust-lang.org/alloc/string/trait.ToString.html#tymethod.to_string
+/// [`core::fmt::Display`]: https://doc.rust-lang.org/core/fmt/trait.Display.html
```

Comment:
```suggestion
/// Note that displaying the type in `sysfs` will fail if
/// [`alloc::string::ToString::to_string`] (as implemented through the
/// [`core::fmt::Display`] trait) writes more than [`PAGE_SIZE`]
/// bytes (including an additional null terminator).
```

Ditto: i.e. the Rust intra-doc links work are found automatically by rustdoc.

I think `PAGE_SIZE` will also work without `kernel::`, but please double-check.

Reply:
Removed the links `rust-lang.org`, but `PAGE_SIZE` needs to be referenced as `crate::PAGE_SIZE`.